unhash: add page (#22388)
tldr-pages/tldr: pages/common/unhash.md

# unhash

> > Remove some cached element from an internal hash table.

Reviewer: ```suggestion
> Remove hashed executable locations.
```

Author: what? it doesnt have to be an executable.

Author: okay i guess thats spliting hairs

> See also: `hash`.

> > More information: <https://fossies.org/linux/zsh/Doc/help/unhash>.

Reviewer: ```suggestion
> More information: <https://zsh.sourceforge.io/Doc/Release/Shell-Builtin-Commands.html>.
```

Author: any suggestions for where to get docs?

Reviewer: I think both of the sources are valid for `unhash`, I suggested the source since it was used in the other zsh builtin pages.

Does this answer your question ? or did you mean something else ??

Author: well i mean my link links directly to unhash but in the new source you have to look for it in the list, and generally i was asking about where to find these sources

Reviewer: For me I just stumble upon the PR comment threads, or while searching for something on the internet. I know this is a vague answer but it just happens and just accrues overtime.

> - Remove a [c]ommand from the hash table:

Reviewer 2: There is no `-c` flag in the command itself.

```suggestion
- Remove a specific command from the hash table:
```

`unhash {{command}}`

- Unhash non-suffix [a]liases:

`unhash -a {{alias}}`

- Unhash [s]uffix aliases:

`unhash -s {{alias}}`

- Unhash shell [f]unctions:

`unhash -f {{function}}`

- Unhash [d]irectories:

`unhash -d {{directory}}`

- Unhash every function that [m]atches a pattern:

> `unhash -f -m "{{git*}}"`

Reviewer: ```suggestion
`unhash -f -m "{{pattern}}"`
```
